### ISSUE TITLE
Relying on Tor's crypto for PQ

### ISSUE THREAD

**Issue by HulaHoopWhonix** [AnemoneLabs/unmessage#22]
The great news is Tor will quantum computing proof before this summer:

https://lists.torproject.org/pipermail/tor-dev/2017-February/011927.html

IMHO it would be great to have all unMessage rely on Tor's native crypto scheme as much as possible to immediately benefit from these changes once they are released.

**Comment by felipedau** [comment]
Thanks @HulaHoopWhonix, that is awesome!

As currently all of unMessage's packets are exchanged via Tor (and consequently "wrapped" by its encryption), I believe that we would automatically benefit from it when it is enabled.

By "rely on Tor's native crypto" you mean that we should drop the encryption layer we currently use?

**Comment by HulaHoopWhonix** [comment]
>By "rely on Tor's native crypto" you mean that we should drop the encryption layer we currently use?

Yes. I am not an expert in crypto implmentations in any way. So my opinion is that if a adversary armed with a quantum computer can impersonate unMessage users because of legacy pre-quantum asymmetric ciphers - its best to drop them and on Hidden Service crypto instead IMHO.

Also it may make things simpler for implementing things such as groupchats

**Comment by felipedau** [comment]
> Yes. I am not an expert in crypto implmentations in any way. So my opinion is that if a adversary armed with a quantum computer can impersonate unMessage users because of legacy pre-quantum asymmetric ciphers - its best to drop them and on Hidden Service crypto instead IMHO.

I am not an expert either and I agree with you that we should aim for post-quantum, but not by removing the "unMessage protocol layer". One of the good things about it is not only that it has the nice properties of Double Ratchet, but that it is independent in a sense that it can basically work on any transport and still allow private and anonymous conversations (well, the transport must also make you anonymous - it would defeat the purpose if peers connected directly to each other) .

I do think that we should take advantage of these technologies and try to find a way to implement them on unMessage so that we also make the protocol quantum computing proof. I honestly do not know if is possible or how complex it would be though.

> Also it may make things simpler for implementing things such as groupchats

You are right, that's certainly something worth considering.

I hope more news about PQ Tor crypto are released soon!

Thanks @HulaHoopWhonix!
